### PR TITLE
Run id column mapping tests in OSS Delta Lake

### PR DESCRIPTION
**Pull request by ebyhr**

### CODE REVIEW

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java [hunk 1]
```diff
@@ -468,7 +373,7 @@ private void testSpecialCharacterColumnNamesWithColumnMappingMode(String mode)
                 " LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "'" +
                 " TBLPROPERTIES (" +
                 " 'delta.columnMapping.mode' = '" + mode + "'," +
-                " 'checkpointInterval' = 3" +
+                " 'delta.checkpointInterval' = 3" +
```
Comment (findinpath) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java:376]: Interesting - Databricks supports both `checkpointInterval` and `delta.checkpointInterval` settings :).

**Review by findinpath — APPROVED** [review]

**Comment by ebyhr** [comment]
CI hit #13199

**Comment by findinpath** [comment]
CI hit #13199 (the failed test is `TestDeltaLakeAlterTableCompatibility.testCommentOnColumn`)

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java [hunk 2]
```diff
@@ -66,21 +67,9 @@ public void testColumnMappingModeNone()
         }
     }
 
-    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_EXCLUDE_73, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS})
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_73, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS}, dataProvider = "columnMappingDataProvider")
```
Comment (Praveen2112) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java:70]: nit: Can we split the commits into two - One would be merging the tests and other would be running the tests for `DELTA_LAKE_OSS` groups.

Comment (ebyhr) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java:70]: If we split into two commits, the 1st commit will lose `DELTA_LAKE_OSS` coverage for name mapping mode.

Comment (Praveen2112) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java:70]: Makes sense. Can we do the reverse operation - like adding `DELTA_LAKE_OSS` for all mappings and then merge the tests.

Comment (ebyhr) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java]: Updated.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java [hunk 3]
```diff
@@ -68,19 +68,7 @@ public void testColumnMappingModeNone()
...
-
-    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_73, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS})
-    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
-    public void testColumnMappingModeName()
-    {
-        testColumnMappingMode("name");
-    }
-
-    private void testColumnMappingMode(String mode)
+    public void testColumnMappingMode(String mode)
```
Comment (Praveen2112) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java:71]: Did we miss the dataProvider ?

**Review by Praveen2112 — APPROVED** [review]

**Pull request merged**